### PR TITLE
Add narrow dendrite check

### PR DESCRIPTION
**Pull request by wizmer**
Add a function for checking that the neuron has no narrow dendrites.
Narrow dendrites are dendrites whose at least one segment greater than a given threshold has a mean radius smaller than a threshold radius.

See https://bbpteam.epfl.ch/project/issues/browse/NSETM-275

### CODE REVIEW

File: neurom/check/neuron_checks.py [hunk 1]
```diff
@@ -287,3 +287,26 @@ def is_dangling(neurite):
...
+        first point
+    '''
+    def dendrite_filter(n):
+        '''Select only dendrites'''
+        return n.type == NeuriteType.basal_dendrite or n.type == NeuriteType.apical_dendrite
+
+    considered_sections = filter(lambda sec: sec.length > considered_section_min_length,
+                                 iter_sections(neuron, neurite_filter=dendrite_filter))
+
+    def bad_section(section):
```
Comment (jdcourcol) [inline neurom/check/neuron_checks.py]: I would call it narrow_section

Comment (wizmer) [reply neurom/check/neuron_checks.py]: What about narrow_dendritic_section ?
This makes me think, should the check also be applied to axons as well ?

Comment (jdcourcol) [reply neurom/check/neuron_checks.py]: I meant replacing "bad_section" by "narrow_section". Then adding _dendritic_ does not bring much as you are not filtering if it is dendritic in that "bad_section" function.

Comment (jdcourcol) [reply neurom/check/neuron_checks.py]: for the axons, I would ask one of the scientists if that is useful before.

Comment (wizmer) [reply neurom/check/neuron_checks.py]: I see. Renamed.

File: neurom/check/neuron_checks.py [hunk 2]
```diff
@@ -287,3 +287,27 @@ def is_dangling(neurite):
     bad_ids = [(neurite.root_node.id, [neurite.root_node.points[1]])
                for neurite in iter_neurites(n) if is_dangling(neurite)]
     return CheckResult(len(bad_ids) == 0, bad_ids)
+
+
+def has_no_narrow_dendritic_section(neuron, radius_threshold=0.05,
```
Comment (jdcourcol) [inline neurom/check/neuron_checks.py]: can we have instead has_no_narrow_neurite_section ?
with a parameter than can select which neurite type are taken into account ?

Comment (wizmer) [reply neurom/check/neuron_checks.py]: Good point. Added

**Pull request merged**